src/schema/formats.js

```diff
@@ -15,7 +15,7 @@ const TOOLKIT_REGEXP =
 
 export function isValidVersionString(version) {
   // We should be starting with a string.
-  if (typeof version !== 'string') {
+  if (typeof version !== 'string' || version.length > 100) {
```

Comment:
Are we happy that there's no version strings that would hit this limit currently? It's unlikely but it would be good to check.

Reply:
Yes we are, I think. The following are the 15 longest version strings we have in our database.

| Length | Version string                                   | add-on slug                        |
| ------ | ------------------------------------------------ | ---------------------------------- |
| 47     | 1.9.2.20110409044302.1-signed.1-signed.1-signed  |    jeteye                          |
| 47     | 1.9.3.20120129113830.1-signed.1-signed.1-signed  |    jeteye                          |
| 46     | 0.4b2.2013051811.1-signed.1-let-fixed.1-signed   |    autoproxy                       |
| 46     | baidu_after_right-click.rev3.1-signed.1-signed   |    baidurightclick                 |
| 46     | 2.4.1+0.1-signed.1-signed.1-signed.1-let-fixed   |    direct-currency-converter-2-fo  |
| 44     | Release_Candidate_1.rev115.1-signed.1-signed     |    symfony2-debug-switch           |
| 44     | Virus_total_scan_url.rev11.1-signed.1-signed     |    virus-total-scan-url            |
| 44     | SimSimi_for_Firefox.rev227.1-signed.1-signed     |    simsimi-for-firefox             |
| 44     | 0.772.1-signed.1-signed.1-let-fixed.1-signed     |    vlcwatcha                       |
| 43     | initial-production.rev267.1-signed.1-signed      |    panther-front-end               |
| 43     | 0.4b2.2015031118.1-signed.1-signed.1-signed      |    autoproxymod                    | 
| 42     | 0.4.3.1Build201012161000.1-signed.1-signed       |    customize-your-web              |
| 42     | 0.5.3.1Build201101051500.1-signed.1-signed       |    mouseless-browsing              |
| 42     | 0.4.3.2Build201102090000.1-signed.1-signed       |    customize-your-web              |
| 42     | 0.4.3.3Build201201122030.1-signed.1-signed       |    customize-your-web              |

Reply:
I think a TODO reminding us to explain why we have made this change is reasonable. 😅

Reply:
Those version strings are not WebExtensions compatible, so `false` will be returned anyway without having to determine the length of the string.
I thought the legacy add-ons are responsible for amo-validator, and addons-linter only need to consider WebExtensions? [Remove support for webextensions in amo-validator · mozilla/amo-validator@cce447e](https://github.com/mozilla/amo-validator/commit/cce447ebe7db90f32cae732f1e5c704ca248b1bc "Remove support for webextensions in amo-validator · mozilla/amo-validator@cce447e")

Reply:
That's true, I just ran the query through all add-ons no matter what to get an idea of how long they get. So 100 is a fairly good limit here.

> I think a TODO reminding us to explain why we have made this change is reasonable. 😅

True, I'll reference the bugzilla bug

Reply:
Indeed, problems can arise if the length of the string exceeds the range of Number.MAX_SAFE_INTEGER, so I can understand if it was `version.length > Number.MAX_SAFE_INTEGER`.
But I do not see much basis for `100`?